src/gtkspellcheck/spellcheck.py

```diff
@@ -481,6 +506,12 @@ def check_range(self, start, end, force_all=False):
                 break
             word_start = word_end.copy()
 
+        if batched_recheck and not end.is_end():
+            start = end.copy()
+            start.forward_char()
+            start_mark = self._buffer.create_mark(None, start)
+            GLib.idle_add(self._continue_batched_recheck, start_mark)
```

Comment:
Yes, definitely! Looking at it again you're absolutely right re `check_range`.

I've made that change but haven't done anything in regards to `buffer_initialize` due to the fact that we need to ensure the batching gets used in other cases where `recheck` is used (eg. changing language or a replacement of the entire buffer while checking is disabled). 

I've made a shuffle, let me know what you think.

I nearly went and catered for the case where the buffer is replaced in the `TextView`, but as that's not otherwise handled anywhere else I ended up not going down that path.

Reply:
Sorry, I confused `buffer_initialize` with `recheck`.

I still wonder what happens if the buffer is replaced while a batched recheck is running? 🤔

Reply:
Yeah.... if the buffer content is replaced I'm pretty sure it's ok. If the `GtkTextBuffer` itself is replaced... hrmm.

In that case `_buffer` is never reassigned anyway, a separate but obviously related problem.

Reply:
It is reassigned in `buffer_initialize`. Actually, `buffer_initialize` is there because it allows the application to swap out the `GtkTextBuffer` of the `GtkTextView` (this is what *hot swapping* in the README refers to). I suggest the following: Whenever `_continue_batched_recheck` is called, it should verify that `_buffer` and the buffer of the start mark are the same. If this is not the case, it simply terminates, no harm done. Otherwise, it triggers the recheck of the specific range.

Reply:
Right, sorry I'd presumed that as there seemed to be no auto-handling of a buffer switch (via a `notify::buffer` on the `TextView`) switching the buffer wasn't catered for. Nice, the change sounds good, I've put it in.